### PR TITLE
scalability framework: separate build for connection workload

### PR DESCRIPTION
**Pull request by nrainer-materialize**
Separate build with higher concurrencies.

Note that I am using a larger agent because the normal one got lost at concurrency 4096. Let me know if this is fine or if I should reduce the maximum concurrency.

### Nightlies
https://buildkite.com/materialize/nightlies/builds?branch=nrainer-materialize%3Ascalability%2Fseparate-connections-build

### CODE REVIEW

**Comment by philip-stoev** [comment]
Those latencies with a high concurrency are just crazy, so care needs to be taken to make sure that they reflect the true performance of Mz and not a problem elsewhere in the stack. I would suggest the following:

1. Benchmark the workloads from `workloads_test.py` and see how scalable are they. If they do not scale, then the bottleneck is on the python/futures side.
2. Try with and without balancerd
3.  (Maybe, depending on the outcome of #1), attempt to benchmark connection speed to some service that is known to have no issue accepting many connections. Maybe a properly-configured Postgres , or some toy or non-toy server that the Internet says is good for this purpose?

**Comment by nrainer-materialize** [comment]
>     1. Benchmark the workloads from `workloads_test.py` and see how scalable are they. If they do not scale, then the bottleneck is on the python/futures side.

Triggered https://buildkite.com/materialize/nightlies/builds/6141.

File: ci/nightly/pipeline.template.yml [hunk 1]
```diff
@@ -101,6 +101,28 @@ steps:
               - common-ancestor
               - --max-concurrency
               - "256"
+    - id: connection-initiation-benchmark
```
Comment (philip-stoev) [inline ci/nightly/pipeline.template.yml:104]: In keeping with the style of the rest of the pipeline, I suggest you create a `scalability-benchmark` group and then the 2 (possibly more in the future) CI steps under it.

Comment (philip-stoev) [reply ci/nightly/pipeline.template.yml:104]: Oh wait, it is already in the "benchmarking" group. So I guess no opportunity for more nesting. Please disregard the comment above.

File: ci/nightly/pipeline.template.yml [hunk 2]
```diff
@@ -101,6 +101,28 @@ steps:
...
+            composition: scalability
+            args:
+              - --target
+              - HEAD
+              - --target
+              - common-ancestor
+              - --regression-against
+              - common-ancestor
+              - --min-concurrency
+              - "256"
```
Comment (philip-stoev) [inline ci/nightly/pipeline.template.yml:121]: I think you still need to benchmark concurrency=1 here . What if connection establishment started taking 1 second individually, but be 100% scalable? We would want Buildkite to detect this and fail.

So maybe just set --min-concurrency to 1 and `--exponent-base` to `4` or `8` ?

File: ci/nightly/pipeline.template.yml (hunk 1 above)
Comment (nrainer-materialize) [reply ci/nightly/pipeline.template.yml:104]: I indeed plan to create three separate jobs:
* scalability with selects
* scalability with connection
* scalability with DDL

File: ci/nightly/pipeline.template.yml (hunk 2 above)
Comment (nrainer-materialize) [reply ci/nightly/pipeline.template.yml]: ✅

**Comment by nrainer-materialize** [comment]
I refactored the whole PR to split workloads into groups. Please re-review.

**Review by def- — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
Nightly reveals some issues with concurrency 4096.
I reduced the concurrency to 2048 for now and **will analyze this and other issues as follow-up.**

**Pull request merged**